### ISSUE TITLE
Runtime error when installing as service with insufficient priviledges from Chocolatey

### ISSUE THREAD

**Issue by Wirone** [microsoftarchive/redis#365]
I have business laptop without admin account access, but with Chocolatey installed with admin priviledges (yeah, weird). When I run `choco install redis-64` it's installed properly, but when I run `redis-server --service-install` I got `HandleServiceCommands: runtime error caught. message=invalid handle passed to constructor`.

Problem exists both in Windows' command line and Cygwin's console.

I ended up with installation from `msi` package with company admins' support, so service is already installed, running and working properly. But I think in that first case message should be different.

**Comment by enricogior** [comment]
Hi @Wirone 
thank you for posting the issue.
I agree, the error message is not user friendly at all, it should give a better indication of the problem that, as you said, it's caused by not having the admin privileges.
Thank you.

**Comment by BigBlueHat** [comment]
@enricogior I just tried using the `redis-server --service-install` command, but hit the same issue...while in a "Run as Administrator" `cmd` instance--the same one I used to run the `choco install redis-64` account in.

Not sure how to get around it. :confused:

**Comment by mattwoberts** [comment]
Same here-  running in a admin command prompt I get this error

`
C:\Windows\system32>redis-server --service-install
[10960] 29 Jan 10:50:58.807 # HandleServiceCommands: system error caught. error code=1073, message = CreateService failed: unknown error
`

**Comment by enricogior** [comment]
@BigBlueHat @mattwoberts 
if you call the `redis-server --service-install` command without passing a configuration file (i.e `redis-server --service-install redis-windows-service.conf`) the working directory by default is the folder where the binaries are, therefore the NETWORK SERVICE account needs to have read and write access to the that folder. Since it doesn't have access to the chocolatey lib folder, the command fails.
In general, if you need to run redis as a service, is preferable to use the MSI installer.

**Comment by colltoaction** [comment]
@enricogior yeah but Chocolatey is great and we'd like to use it! Is there any other way to workaround this?

E.g. I can make it work with:

```
choco install redis-64
cd C:\ProgramData\chocolatey\lib\redis-64\
redis-server --service-install .\redis.windows-service.conf
```

**Comment by enricogior** [comment]
Hi @tinchou 
sorry, I'm not working on this project anymore.

**Comment by DevMostafa** [comment]
The same error when running it out with the full command containing the config file path:

redis-server --service-install .\redis.windows-service.conf

Anybody knew the solution to this error?

**Comment by JesusPuga** [comment]
> The same error when running it out with the full command containing the config file path:
> 
> redis-server --service-install .\redis.windows-service.conf
> 
> Anybody knew the solution to this error?

@DevMostafa  Did u solve it? I have the same error executing the command
